### PR TITLE
Upgrade Modin/pandas and related packages

### LINKED ISSUE #1085: Pandas 2.0 conflicts with the SQL connectors
### System Info

OS version: any
Python version: any
pandasai: any

### 🐛 Describe the bug

Our current setup is facing issues when attempting to upgrade to pandas versions beyond 2.0, as this update seems to interfere with the functionality of our SQL connectors. It's essential for us to delve into the root cause of this problem to identify a viable solution. This will enable us to successfully transition to a more recent version of pandas, as well as to the latest version of modin.

### PR DESCRIPTION
**Pull request by YarShev**
- [x] Closes #1085,  (Replace xxxx with the GitHub issue number).
- [ ] Tests added and passed if fixing a bug or adding a new feature.
- [x] All [code checks passed](https://github.com/gventuri/pandas-ai/blob/main/CONTRIBUTING.md#-testing).

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -10,18 +10,18 @@ packages = [{include = "pandasai"}]
 [tool.poetry.dependencies]
 python = ">=3.9,<3.9.7 || >3.9.7,<4.0"
 python-dotenv = "^1.0.0"
-pandas = "1.5.3"
+pandas = ">=2.0,<3.0"
```
Comment (YarShev) [inline pyproject.toml:13]: This should allow to install latest pandas 2.2.2 but if you look at the lock file, you will see pandas version = "2.1.4". There are probably some cross-dependencies that are not compatible with pandas 2.2.2. Does anybody know what we can do here to get latest pandas?

Comment (YarShev) [reply pyproject.toml:13]: @gventuri, any thoughts?

Comment (YarShev) [reply pyproject.toml:13]: @gventuri, just a friendly reminder

Comment (YarShev) [reply pyproject.toml:13]: @gventuri, any thoughts?

**Comment by andeplane** [comment]
Following this! Currently using pandas-ai in `pyodide` that has locked pandas to 2.2.0.

**Comment by YarShev** [comment]
@gventuri, will Modin be upgraded too?

**Pull request closed without merging**